### PR TITLE
feat: add new header-related rules and remove old rules

### PR DESCRIPTION
**Pull request by padamstx**
This commit adds three new Spectral-style rules:
- authorization-parameter
- accept-parameter
- content-type-parameter
These new rules will flag a header parameter named
'Authorization', 'Accept' or 'Content-Type', respectively.
They replace the following old rules:
- authorization_parameter
- accept_type_parameter
- content_type_parameter

This commit also includes a new custom function
'disallowedHeaderParameter' which is used by each of the new
rules.  The function can be configured with the name of the
header param to flag.

### CODE REVIEW

File: packages/validator/test/cli-validator/tests/config-file-validator.test.js [hunk 1]
```diff
@@ -115,14 +115,17 @@ describe('cli tool - test config file validator', function() {
     const res = configFileValidator(config, chalk);
     const capturedText = getCapturedText(consoleSpy.mock.calls);
 
+    // let text = '';
+    // for (i = 0; i < capturedText.length; i++) {
+    //   text += `[${i.toString()}]: ${capturedText[i]}\n`;
+    // }
+    // console.warn(text);
+
```
Comment (pyrooka) [inline packages/validator/test/cli-validator/tests/config-file-validator.test.js:123]: What about these commented-out lines?

**Review by pyrooka — APPROVED** [review]
LGTM!

File: packages/validator/test/cli-validator/tests/config-file-validator.test.js (hunk 1 above)
Comment (padamstx) [reply packages/validator/test/cli-validator/tests/config-file-validator.test.js:123]: I had added this code while debugging test failures, but then commented it out after I got the tests working again.  I'll remove.

File: packages/ruleset/src/functions/disallowed-header-parameter.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "Required rule configuration property 'headerName' not found."
+    );
+  }
+  return checkHeaderParam(param, path, options.headerName.trim().toLowerCase());
+};
+
+// Return an error if 'param' is a header parameter named '<headerName>'.
+
+function checkHeaderParam(param, path, headerName) {
+  const errors = [];
```
Comment (dpopp07) [inline packages/ruleset/src/functions/disallowed-header-parameter.js:29]: See comment from other PR

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

File: packages/ruleset/src/functions/disallowed-header-parameter.js (hunk 2 above)
Comment (padamstx) [reply packages/ruleset/src/functions/disallowed-header-parameter.js]: done

**Pull request merged**